### PR TITLE
:sparkles: Implement JSON serialization (#12)

### PR DESCRIPTION
**Pull request by janjaali**
Implements functionalities to support the Serializable interface in a functional style and adds a simple JSON stringifier implementation.

Added API's:

* `Forge::jsonFromModel(Model, Serializer)`
* `<T> T::toJson(Serializer)` (for some basic types `<T> T::toJson()`
* `JSON::asString`

Relates to #12.

### CODE REVIEW

**Comment by kittinunf** [comment]
@janjaali Wow! Thanks so much for this. I will take a look :)

File: forge/src/main/kotlin/com/github/kittinunf/forge/Forge.kt [hunk 1]
```diff
@@ -1,8 +1,6 @@
 package com.github.kittinunf.forge
 
-import com.github.kittinunf.forge.core.Deserializable
-import com.github.kittinunf.forge.core.DeserializedResult
-import com.github.kittinunf.forge.core.JSON
+import com.github.kittinunf.forge.core.*
```
Comment (kittinunf) [inline forge/src/main/kotlin/com/github/kittinunf/forge/Forge.kt]: 🙇 Can you please not do star import?

Comment (janjaali) [reply forge/src/main/kotlin/com/github/kittinunf/forge/Forge.kt]: Sure, committed the single name imports instead 👍

**Comment by janjaali** [comment]
@kittinunf any other thoughts/ideas/feedback about that feature?

**Pull request merged**